### PR TITLE
Message capacity as setting

### PR DESCRIPTION
**Pull request by s-axford**
!API CHANGES!
DataMessage and CommandMessage have constructor changes!
Maximum Message Length can now be specified when sending a message via MessageSendingService or when setting up a MessageReceivingService.

I would love to hear how people might want to use the maximum message length, as I can change the system as required. 

All former functions should work the same as before, using a default maximum message length that is defined once in Message.h (currently 255, but we could make this a command line argument for a process), so major changes won't be required outside of changing some message constructors.

Required message constructor changes can be completed as follows:
DataMessage msg(*sender*,*receiver*);
should now read
DataMessage msg = DataMessage(*sender*, *receiver*);

and DataMessage msg(0,0)
can be 
DataMessage msg = DataMessage()

There are also some new constructors that allow for users to specify different maximum message lengths in case a subsystem requires larger messages then 256 bytes.

### CODE REVIEW

File: internal-messaging/message/header/CommandMessage.h [hunk 1]
```diff
@@ -11,18 +11,19 @@ using std::string;
...
         //Builds message from character array output of flatten method
         CommandMessage(char* flat);
+        CommandMessage(char* flat, unsigned int max_size);
 
         CommandMessage(string flat);
+        CommandMessage(string flat, unsigned int max_size);
 
         // Flattens message into a compressed character array that can be parsed by the Message(char* flat) constructor
-        // msg - pointer to char array with a minimum size of 256 bytes.
+        // msg - pointer to char array with a minimum size of *capacity* bytes.
```
Comment (AndrewWay) [inline internal-messaging/message/header/CommandMessage.h]: NOT GOOOD!

File: internal-messaging/message/header/DataMessage.h [hunk 2]
```diff
@@ -14,18 +14,20 @@ static const char* request_break = "-";
...
-
+        DataMessage(char* flat, unsigned int max_size);
+        //Builds message from string output of flatten method
         DataMessage(string flat);
+        DataMessage(string flat, unsigned int max_size);
+
 
         // Flattens message into a compressed character array that can be parsed by the Message(char* flat) constructor
-        // msg - pointer to char array with a minimum size of 256 bytes.
+        // msg - pointer to char array with a minimum size of *capacity* bytes.
```
Comment (AndrewWay) [inline internal-messaging/message/header/DataMessage.h]: !

File: internal-messaging/message/test/message_test2.cc [hunk 3]
```diff
@@ -7,7 +7,8 @@
 #include "PowerKeys.h"
 #include "Identifiers.h"
 using namespace std;
-// THIS IS DEPRECIATED - BETTER TO USE EMULATION 
+// THIS IS VERY DEPRECIATED - BETTER TO USE EMULATION 
+// BEST TO REMOVE THIS FILE?
```
Comment (AndrewWay) [inline internal-messaging/message/test/message_test2.cc]: Nuke

**Review by AndrewWay — APPROVED** [review]

File: internal-messaging/message/header/CommandMessage.h (hunk 1 above)
Comment (s-axford) [reply internal-messaging/message/header/CommandMessage.h]: fixed

File: internal-messaging/message/header/DataMessage.h (hunk 2 above)
Comment (s-axford) [reply internal-messaging/message/header/DataMessage.h]: fixed

File: internal-messaging/message/test/message_test2.cc (hunk 3 above)
Comment (s-axford) [reply internal-messaging/message/test/message_test2.cc]: Removed

**Pull request merged**